### PR TITLE
Promote CallComposite mobile view prop to public

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
* Promote CallComposite mobile view prop to public
* Update storybook documentation to highlight this is portrait only

# Why
* Feature work is complete for this

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/CallComposite.tsx [hunk 1]
```diff
@@ -47,7 +47,6 @@ export type CallCompositeOptions = {
    * Choose to use the composite form optimized for use on a mobile device.
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/CallComposite.tsx:47]: nit: form -> layout?

It took me a while to realize that we are not talking about HTML forms.

File: packages/storybook/stories/CallComposite/CallCompositeDocs.tsx [hunk 2]
```diff
@@ -58,10 +58,11 @@ export const getDocs: () => JSX.Element = () => {
 
       <Heading>Running in a Mobile browser</Heading>
       <Description>
-        CallComposite has been designed for both desktop views and mobile views. The CallComposite does not detect if it
-        is running on mobile device vs desktop, and by default the CallComposite UI is optimized for desktop views. If
-        you are running the CallComposite on a mobile device we recommend making use of the `mobileView` flag. This prop
-        can be set at any time and immediately updates the composite UI to be optimized for a mobile device.
+        CallComposite by default is optimized for desktop views. To provide an optimized mobile experience, you may use
+        the `mobileView` property. Currently this only supports Portrait orientation and not landscape. The
+        CallComposite does not detect if it is running on mobile device vs desktop, instead you must understand if your
```
Comment (anjulgarg) [inline packages/storybook/stories/CallComposite/CallCompositeDocs.tsx:63]: nit: understand -> identify

nit: supply -> update/set

**Review by anjulgarg — APPROVED** [review]

File: packages/storybook/stories/CallComposite/CallCompositeDocs.tsx [hunk 3]
```diff
@@ -58,10 +58,11 @@ export const getDocs: () => JSX.Element = () => {
 
       <Heading>Running in a Mobile browser</Heading>
       <Description>
-        CallComposite has been designed for both desktop views and mobile views. The CallComposite does not detect if it
-        is running on mobile device vs desktop, and by default the CallComposite UI is optimized for desktop views. If
-        you are running the CallComposite on a mobile device we recommend making use of the `mobileView` flag. This prop
-        can be set at any time and immediately updates the composite UI to be optimized for a mobile device.
+        CallComposite by default is optimized for desktop views. To provide an optimized mobile experience, you may use
+        the `mobileView` property. Currently this only supports Portrait orientation and not landscape. The
```
Comment (alcail) [inline packages/storybook/stories/CallComposite/CallCompositeDocs.tsx:62]: nit: landscape -> Landscape (to be consistent with Portrait

**Review by alcail — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/CallComposite.tsx (hunk 1 above)
Comment (JamesBurnside) [reply packages/react-composites/src/composites/CallComposite/CallComposite.tsx:47]: Updated to `Optimizes the composite UI for use on a mobile device.`

File: packages/storybook/stories/CallComposite/CallCompositeDocs.tsx (hunk 2 above)
Comment (JamesBurnside) [reply packages/storybook/stories/CallComposite/CallCompositeDocs.tsx:63]: 👌 updated

File: packages/storybook/stories/CallComposite/CallCompositeDocs.tsx (hunk 3 above)
Comment (JamesBurnside) [reply packages/storybook/stories/CallComposite/CallCompositeDocs.tsx]: updated

Comment (JamesBurnside) [reply packages/storybook/stories/CallComposite/CallCompositeDocs.tsx]: updated

**Pull request merged**